### PR TITLE
URO-164 URO-169 URO-171 Add delete functionality, messaging and basic error handling.

### PR DESCRIPTION
**Pull request by dakotablair**
- [x] URO-164 Narratives can be deleted using the Narrative Control Menu.
  - [x] Update modals to use createPortal.
  - [x] If the action is confirmed to succeed, show a message to the user using a toast notification.
  - [x] If the action fails, show a message to the user using a toast notification. Have distinct messages for when:
    - [x] The app is unable to confirm action success.
- [x] URO-169 Add toast notifications for user messaging.
  - [x] Any app messages may be relayed by a toast notification.
- [x] URO-171 Filter left nav bar by customroles.
- [x] Use better Modals.

### CODE REVIEW

File: src/features/navigator/NarrativeControl/Delete.test.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    expect(buttonDelete).toBeTruthy();
+    if (!buttonDelete) throw Error();
+    await act(async () => buttonDelete.click());
+    expect(buttonDelete.click).toThrow();
+  });
+
+  test('throws an error if the delete request fails.', async () => {
+    fetchMock.mockRejectedValue(null);
+    const { container } = render(
+      <ErrorBoundary FallbackComponent={TestingError} onError={logError}>
```
Comment (dauglyon) [inline src/features/navigator/NarrativeControl/Delete.test.tsx:125]: Is this error boundary needed?

File: src/common/api/authService.ts [hunk 2]
```diff
@@ -45,6 +50,16 @@ export const authApi = baseApi.injectEndpoints({
           },
         }),
     }),
+    getMe: builder.query<AuthResults['getMe'], AuthParams['getMe']>({
+      query: ({ token }) =>
```
Comment (dauglyon) [inline src/common/api/authService.ts:54]: Pull token from app state by default? `store.getState()`

File: src/features/navigator/NarrativeControl/Delete.tsx [hunk 3]
```diff
@@ -1,27 +1,66 @@
...
+  });
+
+  const message = `Deleted narrative ${wsId}.`;
   const deleteNarrativeHandler = async () => {
-    await TODOAddLoadingState();
-    dispatch(deleteNarrative({ wsId: narrativeDoc.access_group }));
+    setUserConfirmation(true);
     modalClose();
+    dispatch(deleteNarrative({ wsId }));
+    try {
```
Comment (dauglyon) [inline src/features/navigator/NarrativeControl/Delete.tsx:35]: Use and/or modify this func to solve this section? https://github.com/kbase/ui/blob/8312ed5485651ff3c76ffd1bcd9c03d72552e692/src/common/api/utils/parseError.ts#L4

**Review by dauglyon — APPROVED** [review]

File: src/features/navigator/NarrativeControl/Delete.test.tsx (hunk 1 above)
Comment (dakotablair) [reply src/features/navigator/NarrativeControl/Delete.test.tsx]: Nope!

File: src/features/navigator/NarrativeControl/index.tsx [hunk 4]
```diff
@@ -27,8 +28,9 @@ interface ControlLatestProps {
 }
 
 const ControlLatest: FC<ControlLatestProps> = ({ narrativeDoc }) => {
-  const { getModalControls, setModalContents } = useContext(ModalContext);
-  const { modalClose } = getModalControls();
+  const [modalView, setModalView] = useState('Copy this Narrative');
+  const modal = useModalControls();
+  const modalClose = () => modal?.close();
   const { version } = narrativeDoc;
   const controlLatestDialogs: Record<string, JSX.Element> = {
```
[the remark this answers is not in the document]
Comment (dakotablair) [reply src/features/navigator/NarrativeControl/index.tsx]: Done!

File: src/common/api/authService.ts (hunk 2 above)
Comment (dakotablair) [reply src/common/api/authService.ts]: This causes a circular dependency, so I'm going to leave it for now and we can return to this later.

File: src/features/navigator/NarrativeControl/Delete.tsx (hunk 3 above)
Comment (dakotablair) [reply src/features/navigator/NarrativeControl/Delete.tsx:43]: Sure!

**Pull request merged**